reworked the second vignette (#48)
trendecon/trendecon: vignettes/daily-series.Rmd

)
```

Once the combination of useful keywords is selected we might want to use them as economic indicators in production and display them on [trendecon.org](https://www.trendecon.org/). This vignette describes the steps for adding new indicators to the production processes of "Trendecon".

## Directory structure ##

The names of production functions within the `trendecon` package start with `proc_`.
They usually operate on the file system which means that calling these functions will write files to disk.
The `proc_` functions will save files within two folders: `data` and `raw`, which are located under the working directory (`getwd()`).

> `raw` folder holds all the data downloaded from Google, plus some transformations, such as seasonally adjusted or combined series.

Reviewer: ```suggestion
`raw` folder holds all the data downloaded from Google that are automatically transformed by the `trendecon`package. The transformations include seasonal adjustments or combining multiple keywords related to the indicator.
```

Reviewer: I see! So there are the raw time series, and on top some with transformations if called. My bad. I think the sentence you had written is good.

`data` folder, on the other hand, collects the final indicators - the data displayed on the website.
If not present these folders will be created when calling `proc_keyword_init` for the first time.

Within these folders, every used `geo` location will have its own subfolder.
Thus, in the end, if we used Swiss and Austrian indicators, we would end up with the following directory structure:

```
│
├── data
│   ├── at
│   └── ch
│
└── raw
    ├── at
    └── ch
```

## Initial download ##

To include a new series, each keyword must first be initiated.
Our goal is to produce long daily time series, ideally from 2006.
However, Google does not provide daily or weekly data for such a long time period.
Hence we need to circumvent the problem by applying a moving window of daily and weekly queries over the whole time period.

> So be careful, as this causes a lot of queries to Google.

Reviewer: This line made me insecure. You could add, why we should careful (blocking by the API?And yes, for how long?) Or could we delete it?

Reviewer: I like the small addition!

In the first step, the raw data series for each keyword are updated with the latest daily, weekly, and monthly data.
For example, to update the raw series for keyword `"Rezession"` for Switzerland, the script calls the following internal function:

```r
proc_keyword_latest("Rezession", "CH")
```

This function downloads raw daily, weekly, and monthly data for the specified pair of keyword and geo location.

### Step 2: bending ###

To combine the three frequencies (monthly, weekly, and daily), we apply the following methodology: in a first step, we "bend" the daily series to the weekly values, by applying a variant of the Chow-Lin (1971) method.
This preserves the movement of the daily series and ensures that weekly averages are identical to the original weekly series.
We then use the same methodology to bend the series to the monthly values.

To combine the three frequencies for a given keyword, the script calls another internal function:

```r
proc_combine_freq("Rezession", "CH")
```

### Step 3: seasonal adjustment ###

Some keywords might display seasonal patterns.
For example it is not surprising that searches for gardening are higher in spring.

> In order to make meaningful comparisons over time such seasonal patterns, present within the data, need to be removed.

Reviewer: ```suggestion
In order to make meaningful comparisons over time such seasonal patterns, present within the raw data, need to be filtered out.
```

To achieve this `trendecon` uses the "Prophet" procedure for estimating an additive model where non-linear trends are fit with yearly and weekly seasonality and the holiday effects.

To seasonally adjust a combined keyword, the script calls the following internal function:

```r
proc_seas_adj("Rezession", "CH")
```

### Step 4: combining ###

> Once the data for each keyword has been processed we end up with multiple time series - one for each keyword within the indicator.

Reviewer: ```suggestion
Once the raw data for each keyword has been processed we end up with multiple time series - one for each keyword within the indicator.
```

As an example the main indicator uses the following keywords: `"Wirtschaftskrise"`, `"Kurzarbeit "`, `"arbeitslos"`, and `"Insolvenz"`.
To turn several provided keywords into a single time series the first principal component is used.

In order to achieve this the script calls `ts_prcomp()` function from the `tsbox` package.

### Step 5: writing the data ###

> Finally the prepared index is saved to a file:

Reviewer: ```suggestion
Finally the prepared index is saved to a file in the data folder:
```

```
write_keyword(prepared_data, "indicator", "CH")
```

## Summary ##

In summary:

> - To add new indicator "indicator": create a new script at `inst/script/indicator.R`.

Reviewer: ```suggestion
- To add a new indicator "indicator": create a new script at `inst/script/indicator.R`.
```

```

This function downloads raw daily, weekly, and monthly data for the specified pair of keyword and geo location.

> If the data for a particular keyword is not yet available, `proc_keyword_init()` should be called instead.

Reviewer: the function downloaded time series (d,w,m) for the keyword "Insvolvenz". The function above, proc_keyword_latest I could not execute "could not find function "proc_keyword_latest". I guess this is to come?

Reviewer: great! thanks for the explanation and hint!
